### PR TITLE
Null checks, Prettier user-end appearance, etc

### PR DESCRIPTION
**Pull request by AbandonedCart**

### CODE REVIEW

**Comment by AbandonedCart** [comment]
This avoids changing any values until it can be verified how they were originally determined. It's been 4 years since they were originally implemented and the change was picked from a remote that is no longer stored on Github. For now, given the limited usage, they are being left as-is.

**Comment by AbandonedCart** [comment]
The above change will have a single use today, but the base concept can also be leveraged for a lot of other things in the future.

File: shell/android-studio/reicast/src/main/java/com/reicast/emulator/config/OptionsFragment.java [hunk 1]
```diff
@@ -767,4 +764,34 @@ private void showToastMessage(String message, int duration) {
 				.getDimensionPixelOffset(R.dimen.snackbar_icon_padding));
 		snackbar.show();
 	}
+
+	private int getBroadcastValue(String broadcastName) {
+		if (broadcastName.equals("NTSC-J"))
```
Comment (baka0815) [inline shell/android-studio/reicast/src/main/java/com/reicast/emulator/config/OptionsFragment.java]: Isn't that something that should be done inside reicast core to avoi duplicate code?

File: shell/android-studio/reicast/src/main/java/com/reicast/emulator/Emulator.java [hunk 2]
```diff
@@ -107,6 +107,28 @@ public void loadGameConfiguration(String gameId) {
         JNIdc.bootdisk(mPrefs.getString(pref_bootdisk, bootdisk));
     }
 
+    public void isVGACompatible(String gameId) {
```
Comment (baka0815) [inline shell/android-studio/reicast/src/main/java/com/reicast/emulator/Emulator.java]: Same as my other comment.

File: shell/android-studio/reicast/src/main/java/com/reicast/emulator/config/OptionsFragment.java (hunk 1 above)
Comment (AbandonedCart) [reply shell/android-studio/reicast/src/main/java/com/reicast/emulator/config/OptionsFragment.java]: It's only used in two places and under review. If the values are incorrect, the replacement code does not require any sort of helper. It was done this way to make for a simple removal.

File: shell/android-studio/reicast/src/main/java/com/reicast/emulator/Emulator.java (hunk 2 above)
Comment (AbandonedCart) [reply shell/android-studio/reicast/src/main/java/com/reicast/emulator/Emulator.java]: This is as general as it gets. This class extends Application.

**Comment by AbandonedCart** [comment]
Important note: This only includes NA / EU releases. The forum hints to a list of NTSC-J games, but it was not linked and will take some digging to determine if it was ever made. Ideally `-1` will just default to the user setting.

**Comment by AbandonedCart** [comment]
Switched the values to favor patchable over TV-only, since that is an "upgrade"

**Comment by AbandonedCart** [comment]
This prevents the user from being forced into a setting, but still guides them to a default based on what testing has proven is compatible.

**Comment by AbandonedCart** [comment]
```
                PAL-E - Conflict Zone - GOT - [DCCM]
                NTSC-U - Donald Duck: Goin' Quackers - GOT - [DCCM/RDC]
                NTSC-U - ESPN International Track n Field - GOT -[DCCM]
                PAL-E - Millenium Soldier: Expendable (Expendable) - GOT - [DCP]
                PAL-E - Heavy Metal: Geomatrix - GOT - [DCP]
                NTSC-U - House of the Dead 2 - GOT - [DCCM]
                NTSC-U - Loony Toons Space Race - GOT - [DCRES/OVERRiDE]
                PAL-E - Midway Greatest Arcade Hits Volume 1 - GOT - [DCCM]
                PAL-E - Mortal Kombat Gold - GOT - [DCP]
                PAL-E - NBA Hoopz - GOT - [PULSAR]
                NTSC-U - NHL 2K - GOT - [DCCM]
                PAL-E Freestyle Scooter [Razor Freestyle Scooter]
                NTSC-U Record of Lodoss War
                NTSC-U Roadsters
                PAL-E San Fransisco Rush 2049
                NTSC-U Sega Bass Fishing
                NTSC-U Sega Rally Championship 2
                NTSC-U Silent Scope
                PAL-E Speed Devils Online Racing
                PAL-E Star Wars: Demolition
                PAL-E Star Wars: Episode 1 Jedi Power Battles
                PAL-E Star Wars: Episode 1 Racer
                PAL-E Street Fighter III: 3rd Strike
                PAL-E Suzuki Alstare Extreme Racing
                PAL-E Tech Romancer
                PAL-E Tokyo Highway Challenge 2 [Tokyo Extreme Racer 2]
                NTSC-U Toy Comander
                NTSC-U Toy Story 2: Buzz Lightyear to the Rescue!
                PAL-E TrickStyle
                PAL-E Urban Chaos
                NTSC-U Vigilante 8: 2nd Offense
                NTSC-U Wetrix+
                NTSC-U Xtreme Sports
                NTSC-U Zombie Revenge
```

This is the list of games (from the comments) that were never tested. IDs will be added once the current progress has been verified.

**Comment by AbandonedCart** [comment]
For simplicity, this now returns 3 (TV / emu default) for unknown. No use returning -1 just to return an alternate value, only to read it as 3 anyway.

**Comment by AbandonedCart** [comment]
@flyinghead I believe you use a hybrid version of the per-game settings that sets the default based on the game, but then allows overriding it with a configured value. I think this should match Android up to much what you were already doing with the addition of the VGA hoopla.
I know you mentioned there being some games that required certain settings, so hopefully the testing done for the list had the same results.

**Comment by baka0815** [comment]
> This is the list of games (from the comments) that were never tested.
> They are currently set to default, but could be updated in the future.

What would be necessary to confirm they work?

**Comment by AbandonedCart** [comment]
Set the cable to VGA and see if the graphics still work.

**Comment by AbandonedCart** [comment]
Adding unconfirmed items to a final list and converting this for other platforms can be a follow-up. This is already getting a bit lengthy. Assuming the toast are not displaying because they had a limited context, this should also address the discussion of how to notify the user about CDI in https://github.com/reicast/reicast-emulator/issues/1410

**Comment by flyinghead** [comment]
@AbandonedCart Yes, I originally backported libretro per-game settings, which are hard-coded and non overridable I believe. With the PGC in emu.cfg and in the android UI, there is probably redundant stuff. More technical debt...

**Comment by AbandonedCart** [comment]
I think it's actually more effective that way. Set a values based on what we know, but let the user change it if they disagree (or some other factor comes into play)

**Pull request merged**